### PR TITLE
Feat/welcome guide popover and carousel

### PR DESCRIPTION
**Pull request by isha-dubey**
## Description

#962 
#963 
#964 
#965 

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build

## Have you tested locally?

- [ ] 👍 yes
- [ ] 🙅 no, because I am lazy

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md

### CODE REVIEW

File: src/packages/@app/models/guide.model.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    id: {
+      type: "string",
+      maxLength: 100,
+    },
+    isActive: {
+      type: "boolean",
+    },
+  },
+  required: ["id"],
+} as const;
```
Comment (itsmdasifraza) [inline src/packages/@app/models/guide.model.ts:22]: also use "isActive" in  required field

File: src/packages/@app/pages/Collections/CollectionPage.ViewModel.ts [hunk 2]
```diff
@@ -2399,4 +2402,23 @@ export default class CollectionsViewModel {
...
+    return await this.guideRepository.findOne({
+      id: "collection-guide",
+    });
+  };
+  public updateCollectionGuide = async (isActive: boolean) => {
+    await this.guideRepository.update(
+      {
+        id: "collection-guide",
+      },
+      {
```
Comment (itsmdasifraza) [inline src/packages/@app/pages/Collections/CollectionPage.ViewModel.ts:2418]: make this generic, pass id as an argument and use as a parameter

File: src/packages/@app/pages/Collections/CollectionPage.ViewModel.ts [hunk 3]
```diff
@@ -2399,4 +2402,23 @@ export default class CollectionsViewModel {
...
+        isActive: isActive,
+      },
+    );
+    console.log(isActive);
+  };
```
Comment (itsmdasifraza) [inline src/packages/@app/pages/Collections/CollectionPage.ViewModel.ts:2423]: logs not required

File: src/packages/@app/pages/EnvironmentExplorer/EnvironmentExplorerPage.ViewModel.ts [hunk 4]
```diff
@@ -153,4 +156,22 @@ export class EnvironmentExplorerViewModel {
...
+  };
+  public updateEnvironmentGuide = async (isActive: boolean) => {
+    await this.guideRepository.update(
+      {
+        id: "environment-guide",
+      },
+      {
+        isActive: isActive,
+      },
+    );
```
Comment (itsmdasifraza) [inline src/packages/@app/pages/EnvironmentExplorer/EnvironmentExplorerPage.ViewModel.ts:175]: same here and  add js doc comments

File: src/packages/@app/repositories/guide.repository.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    const foundDocs = await rxCollection
+      .findOne({
+        selector: query,
+      })
+      .exec();
+
+    await foundDocs.incrementalModify((value) => {
+      return { ...value, ...updateData };
+    });
+  };
```
Comment (itsmdasifraza) [inline src/packages/@app/repositories/guide.repository.ts:40]: js doc comment required

File: src/packages/@common/components/sidebar/Sidebar.svelte [hunk 6]
```diff
@@ -106,14 +107,16 @@
...
-    </div>
+  <div class="secondary-sidebar-items z-4">
+    {#each secondarySidebarItems as item (item.route)}
+      <SidebarItem {item} />
+    {/each}
+    <Tooltip title="User Profile" placement="right" distance={20} >
+    <SidebarProfileModal item={sidebarModalItem} {onLogout} />
+    </Tooltip>
   </div>
 </div>
```
Comment (itsmdasifraza) [inline src/packages/@common/components/sidebar/Sidebar.svelte:118]: can we create a separate pull request for this?

File: src/packages/@deprecate/components/environments/enviroments-panel/EnvironmentPanel.svelte [hunk 7]
```diff
@@ -20,6 +20,7 @@
     workspaceLevelPermissions,
   } from "$lib/utils/constants/permissions.constant";
   import Tooltip from "@library/ui/tooltip/Tooltip.svelte";
+    import WelcomeBox from "@library/ui/popover/Popover.svelte";
 
   export let environmentRepositoryMethods: EnvironmentRepositoryMethods;
```
Comment (itsmdasifraza) [inline src/packages/@deprecate/components/environments/enviroments-panel/EnvironmentPanel.svelte:25]: import this from "@library/ui" only, and push this component to index.ts

File: src/packages/@deprecate/components/environments/enviroments-panel/EnvironmentPanel.svelte [hunk 8]
```diff
@@ -186,6 +187,9 @@
     {/if}
   </div>
 {/if}
+<div>
+  <WelcomeBox/>
+</div>
```
Comment (itsmdasifraza) [inline src/packages/@deprecate/components/environments/enviroments-panel/EnvironmentPanel.svelte:193]: can we have a better name for this?

File: src/packages/@deprecate/utils/interfaces/request.interface.ts [hunk 9]
```diff
@@ -117,6 +117,9 @@ export interface Save {
 }
 
 export interface Request {
+  source: string;
+  id(id: any, arg1: { items: any; collectionId: any; workspaceId: string });
+  name: string;
```
Comment (itsmdasifraza) [inline src/packages/@deprecate/utils/interfaces/request.interface.ts:122]: is this required?

File: src/packages/@deprecate/utils/interfaces/request.interface.ts [hunk 10]
```diff
@@ -139,6 +142,9 @@ export interface WorkSpace {
 }
 
 export interface Folder {
+  source: string;
+  id: any;
+  name: any;
   requestCount?: number;
   folderCount?: number;
```
Comment (itsmdasifraza) [inline src/packages/@deprecate/utils/interfaces/request.interface.ts:149]: is this required?

File: src/packages/@environments/features/environment-explorer/layout/EnvironmentExplorer.svelte [hunk 11]
```diff
@@ -12,6 +13,12 @@
   import { TabularInput } from "@environments/common/components";
   import { WithButton } from "@environments/common/hoc";
   import { Input } from "@library/forms";
+  import { Carousel, Modal, Popover } from "@library/ui";
+  import {
+    CreateENV,
+    IntroToEnvironment,
+    SearchVariable,
+  } from "@environments/features";
```
Comment (itsmdasifraza) [inline src/packages/@environments/features/environment-explorer/layout/EnvironmentExplorer.svelte:21]: do not import these gifs from environment/features. use "../videos" instead

File: src/packages/@environments/features/environment-explorer/layout/EnvironmentExplorer.svelte [hunk 12]
```diff
@@ -30,10 +37,13 @@
    */
   export let onSaveEnvironment;
 
+  export let onFetchEnvironmentGuide;
+  export let onUpdateEnvironmentGuide;
+  let showContainer = false;
+
```
Comment (itsmdasifraza) [inline src/packages/@environments/features/environment-explorer/layout/EnvironmentExplorer.svelte:43]: can we have a better name for that clarifies what type of container it is?
also use "is" before variable name.

File: src/packages/@environments/features/environment-explorer/layout/EnvironmentExplorer.svelte [hunk 13]
```diff
@@ -49,12 +59,43 @@
   ) => {
     onUpdateVariable(pairs);
   };
+  let isGuidePopup = false;
+
+  onMount(async () => {
+    const event = await onFetchEnvironmentGuide();
+    console.log(event);
+    if (event.isActive === true) {
+      showContainer = true;
```
Comment (itsmdasifraza) [inline src/packages/@environments/features/environment-explorer/layout/EnvironmentExplorer.svelte:68]: logs not required

File: src/packages/@environments/features/environment-explorer/layout/EnvironmentExplorer.svelte [hunk 14]
```diff
@@ -49,12 +59,43 @@
...
+              if (showContainer === true) {
+                onUpdateEnvironmentGuide(true);
+              } else {
+                onUpdateEnvironmentGuide(false);
+              }
+            }}
+          >
+            <HelpIcon height={"12.67px"} width={"12.67px"} />
+          </button>
+        {/if}
```
Comment (itsmdasifraza) [inline src/packages/@environments/features/environment-explorer/layout/EnvironmentExplorer.svelte:97]: could you please confirm with the design team about this, I think we have to show this for both local and global environment

File: src/packages/@library/ui/popover/Carousel.svelte [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+    if (currentStep > 0) {
+      currentStep = currentStep - 1;
+      stepData = data[currentStep];
+    }
+  }
+
+  function close() {
+    alert("hello");
+  }
+</script>
```
Comment (itsmdasifraza) [inline src/packages/@library/ui/popover/Carousel.svelte:26]: ??

File: src/packages/@library/ui/popover/Carousel.svelte [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+      stepData = data[currentStep];
+    }
+  }
+
+  function previousStep() {
+    if (currentStep > 0) {
+      currentStep = currentStep - 1;
+      stepData = data[currentStep];
+    }
+  }
```
Comment (itsmdasifraza) [inline src/packages/@library/ui/popover/Carousel.svelte:21]: can we create a navigator function that handles next and prev

File: src/packages/@library/ui/popover/Carousel.svelte [hunk 17]
```diff
@@ -0,0 +1,92 @@
...
+      </div>
+      <header
+        class="d-flex"
+        style=" height:auto; 	justify-content: space-between;"
+      >
+        <div>
+          {#if currentStep > 0}
+            <button
+              class="border-0"
+              style=" background-color: var(--bg-primary-300) ; padding:3px 19px 3px 20px; border-radius:4.78px; outline:none; "
```
Comment (itsmdasifraza) [inline src/packages/@library/ui/popover/Carousel.svelte:65]: rename your parent directory to "carousel" from "popover"

File: src/packages/@library/ui/popover/IntroToEnvironment.gif
Comment (itsmdasifraza) [inline src/packages/@library/ui/popover/IntroToEnvironment.gif:1]: gif at wrong place push this file to environment module

File: src/packages/@library/ui/guidepopup/GuidePopup.svelte [hunk 18]
```diff
@@ -0,0 +1,7 @@
+<script lang="ts">
+
+</script>
+
+<div>
+    hello
+</div>
```
Comment (itsmdasifraza) [inline src/packages/@library/ui/guidepopup/GuidePopup.svelte:7]: this file not required right? and also the parent directory

File: src/packages/@library/ui/index.ts [hunk 19]
```diff
@@ -9,3 +9,4 @@ export * from "./options";
 export * from "./label";
 export * from "./tooltip";
 export * from "./modal";
+export * from "./popover";
```
Comment (itsmdasifraza) [inline src/packages/@library/ui/index.ts:12]: add here a new export for Carousel also

currently you have pushed "carousel and popover" both in "popover" directory, create a separate directory for "carousel"

File: src/packages/@app/pages/Collections/CollectionPage.ViewModel.ts [hunk 20]
```diff
@@ -2384,6 +2387,28 @@ export default class CollectionsViewModel {
...
+  public fetchCollectionGuide = async (query) => {
+    const data = await this.guideRepository.findOne(query);
+    return data;
+  };
+
+  /**
+   * Updates the collection guide document's active status.
+   *
+   * @param {boolean} isActive - The new active status to set for the collection guide.
+   * @returns {Promise<void>} - A promise that resolves when the update is complete.
```
Comment (itsmdasifraza) [inline src/packages/@app/pages/Collections/CollectionPage.ViewModel.ts:2404]: no need to give data type in js doc curly braces, as we already using typescript

File: src/packages/@workspaces/features/workspace-default/layout/WorkspaceDefault.svelte [hunk 21]
```diff
@@ -1,50 +1,74 @@
 <script>
-  import {VectorIcon , LibraryIcon} from "@library/icons";
-    import SparrowLogo from "@workspaces/features/rest-explorer/assets/images/sparrow-logo.svelte";
+  import { VectorIcon, LibraryIcon } from "@library/icons";
+  import SparrowLogo from "@workspaces/features/rest-explorer/assets/images/sparrow-logo.svelte";
+  import { Popover, Modal, Carousel } from "@library/ui";
+  import { onMount } from "svelte";
+  import CollectionsViewModel from "@app/pages/Collections/CollectionPage.ViewModel";
   export let showImportCollectionPopup;
   export let onItemCreated;
```
Comment (itsmdasifraza) [inline src/packages/@workspaces/features/workspace-default/layout/WorkspaceDefault.svelte:8]: we cant import collectionsViewModel at workspacedefault.svelte

File: src/packages/@library/ui/carousel/Carousel.svelte [hunk 22]
```diff
@@ -0,0 +1,83 @@
+<script lang="ts">
+  import Ellipse from "@library/icons/Ellipse.svelte";
+  export let data = [];
```
Comment (itsmdasifraza) [inline src/packages/@library/ui/carousel/Carousel.svelte:3]: import using library/icons only

File: src/packages/@library/icons/Ellipse.svelte [hunk 23]
```diff
@@ -1,3 +1,12 @@
-<svg width="11" height="11" viewBox="0 0 11 11" fill="none" xmlns="http://www.w3.org/2000/svg">
-<circle cx="5.5" cy="5.5" r="5.5" fill="#3670F7"/>
+<script lang="ts">
+    export let width = "10px";
+    export let height = "10px";
+    export let color = "grey";
+  </script>
+<!-- <script>
+    let color = "#3670F7"
+</script> -->
```
Comment (itsmdasifraza) [inline src/packages/@library/icons/Ellipse.svelte:8]: ??

File: src/packages/@workspaces/features/index.ts [hunk 24]
```diff
@@ -6,4 +6,5 @@ export * from "./import-curl";
 export * from "./collection-explorer";
 export * from "./folder-explorer";
 export * from "./tab-bar";
-export * from "./workspace-default"; 
\ No newline at end of file
+export * from "./workspace-default"; 
+export * from "./videos";
```
Comment (itsmdasifraza) [inline src/packages/@workspaces/features/index.ts:10]: export * from "./videos"; 

we will not export videos from /features

File: src/packages/@app/pages/RestExplorer/RestExplorerPage.ViewModel.ts [hunk 25]
```diff
@@ -1077,6 +1079,14 @@ class RestExplorerViewModel
       return response;
     }
   };
+  public fetchCollectionGuide = async (query) => {
+    return await this.guideRepository.findOne(query);
+  };
+  public updateCollectionGuide = async (query, isActive: boolean) => {
```
Comment (Astitva877) [inline src/packages/@app/pages/RestExplorer/RestExplorerPage.ViewModel.ts:1085]: can we add jsdoc here?

**Review by Astitva877 — APPROVED** [review]

**Review by gc-codes — APPROVED** [review]

**Comment by shreyas070707** [comment]
welcome text is working @sanilnayak  please check and close it

**Pull request merged**